### PR TITLE
fix(components/ActionBar): accept children

### PR DESCRIPTION
**Pull request by jmfrancois**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)

The ActionBar component do not accept children.
on each render the action bar create bunch of functions.

**What is the new behavior?**

The ActionBar accept children using custom component.
It place the children using left and right property.
You can use tag props with :
* form
* p
* a
* button

Without tag it's just a div so you can what ever you want. PLease check the storybook.

It has been complelty refactored to just used component.
Should be faster and it's cleaner.
It also attach all sub component so you can create your own if you want

screen shot 2017-04-06 at 11 34 48

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration
path for existing applications: ...

**Other information**:

this is the new way to build component so please check the code as an example we have put lots of effort in it.

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://352.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://352.talend.surge.sh)

File: packages/components/src/ActionBar/ActionBar.component.js [hunk 1]
```diff
@@ -8,90 +8,104 @@ const DISPLAY_MODES = {
...
-	multiSelectActions: PropTypes.shape(actionsShape),
-};
-
-ActionBar.defaultProps = {
-	actions: [],
+	children: PropTypes.node,
 };
 
 ActionBar.DISPLAY_MODES = DISPLAY_MODES;
+ActionBar.Count = Count;
```
Comment (charlesnguyen) [inline packages/components/src/ActionBar/ActionBar.component.js]: Why?

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://352.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://352.talend.surge.sh)

File: packages/components/src/ActionBar/ActionBar.component.js (hunk 1 above)
Comment (jmfrancois) [inline packages/components/src/ActionBar/ActionBar.component.js]: This has been exposed to previous TAB.
it's to help you to build your own custom ActionBar if you need in your project by composition.
So all sub components are attached to the main one.

the idea is coming from react-bootstrap which is doing that.

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://352.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://352.talend.surge.sh)

File: packages/components/stories/ActionBar.js [hunk 2]
```diff
@@ -201,4 +201,35 @@ storiesOf('ActionBar', module)
...
+	))
+	.addWithInfo('custom', () => (
+		<nav>
+			<IconsProvider defaultIcons={icons} />
+			<div id="default">
+				<ActionBar>
+					<ActionBar.Content tag="a" left href="#/foo/bar">
+						Hello anchor
+					</ActionBar.Content>
+					<ActionBar.Content tag="button" className="btn btn-default" left>
```
Comment (charlesnguyen) [inline packages/components/stories/ActionBar.js]: it would be nice to have onClick={action('onClick')} for this tag

Comment (jmfrancois) [inline packages/components/stories/ActionBar.js]: all other props should be passed to the button so it should be already the case

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://352.talend.surge.sh)

**Pull request merged**